app/models/generic_event/per_medical_aid.rb

```diff
@@ -0,0 +1,74 @@
+class GenericEvent
+  class PerMedicalAid < GenericEvent
+    LOCATION_ATTRIBUTE_KEY = :location_id
+    DETAILS_ATTRIBUTES = %w[
+      advised_at
+      advised_by
+      treated_at
+      treated_by
+      supplier_personnel_number
+      vehicle_reg
+    ].freeze
+
+    include LocationValidations
+    include PersonEscortRecordEventValidations
+    include SupplierPersonnelNumberValidations
+
+    validates :advised_at, presence: true
+    validates :advised_by, presence: true
+    validates :treated_at, presence: true
+    validates :treated_by, presence: true
+
+    validates_each :advised_at do |record, attr, value|
+      Time.zone.iso8601(value)
+    rescue ArgumentError
+      record.errors.add(attr, 'must be formatted as a valid ISO-8601 date-time')
+    end
+
+    validates_each :treated_at do |record, attr, value|
+      Time.zone.iso8601(value)
+    rescue ArgumentError
+      record.errors.add(attr, 'must be formatted as a valid ISO-8601 date-time')
+    end
+
+    def advised_at=(advised_at)
```

Comment:
Hmm... these events have a very common pattern for getters and setters with the details hash - seems like a perfect use case for some meta programming which would remove all of these boiler plate getters and setters 🤔

Reply:
I was having the same thought process, actually. I'll look at sorting this in another PR :).